### PR TITLE
feat(new-widget-builder-experience): Add dashboard widget query fieldAliases serializer -  [DD-631] 

### PR DESCRIPTION
**Pull request by priscilawebdev**
- [x]  Add dashboard widget query fieldAliases serializer
- [x]  Add tests

### CODE REVIEW

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py [hunk 1]
```diff
@@ -288,6 +297,40 @@ def test_add_widget(self):
...
+            "title": "First dashboard",
+            "widgets": [
+                {
+                    "title": "Errors per project",
+                    "displayType": "table",
+                    "interval": "5m",
+                    "queries": [
+                        {
+                            "name": "Errors",
+                            "fields": [],
```
Comment (narsaynorath) [inline tests/sentry/api/endpoints/test_organization_dashboard_details.py:311]: We should probably improve the validation to keep the length of fields and field aliases the same. Can we do a follow up PR for that?

File: src/sentry/api/serializers/rest_framework/dashboard.py [hunk 2]
```diff
@@ -115,6 +115,9 @@ class DashboardWidgetQuerySerializer(CamelSnakeSerializer):
         child=serializers.CharField(), required=False, allow_null=True
     )
     columns = serializers.ListField(child=serializers.CharField(), required=False, allow_null=True)
+    field_aliases = serializers.ListField(
+        child=serializers.CharField(), required=False, allow_null=True
```
Comment (narsaynorath) [inline src/sentry/api/serializers/rest_framework/dashboard.py:119]:
```suggestion
        child=serializers.CharField(allow_blank=True), required=False, allow_null=True
```
Realized that this won't let us set empty strings if a field doesn't have an alias. Can we add a test to cover this case as well?

File: tests/sentry/api/endpoints/test_organization_dashboards.py [hunk 3]
```diff
@@ -578,6 +578,43 @@ def test_add_widget_with_invalid_limit_below_minimum(self):
         assert response.status_code == 400
         assert b"Ensure this value is greater than or equal to 1" in response.content
 
+    def test_add_widget_with_fieldAliases_succeeds(self):
```
Comment (narsaynorath) [inline tests/sentry/api/endpoints/test_organization_dashboards.py:581]:
```suggestion
    def test_add_widget_with_field_aliases_succeeds(self):
```
I know the key is actually camel-case, but we should probably use snake case for the title

**Review by narsaynorath — APPROVED** [review]
Just had some small comments around the functionality. I disabled the auto-merge because I think we should include the `allow_blank=True` change in this, but once those changes are in then it won't error when using it with the new design

Tested this by merging in master locally, saved aliases and confirmed they're persisted

File: tests/sentry/api/endpoints/test_organization_dashboard_details.py (hunk 1 above)
Comment (priscilawebdev) [reply tests/sentry/api/endpoints/test_organization_dashboard_details.py:311]: oh yes... sure thing

Comment (priscilawebdev) [reply tests/sentry/api/endpoints/test_organization_dashboard_details.py:311]: I thought about it too but need to clarify some doubts first... will create a draft PR 😉

**Pull request merged**